### PR TITLE
isom-1689 polish for component editing panels

### PR DESCRIPTION
**Pull request by adriangohjw**
## Problem

Closes https://linear.app/ogp/issue/ISOM-1689/polish-for-component-editing-panels except the tiptap editor changes which will handled in a separate PR due to the complexity of having to overwrite tiptap default settings

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible

**Improvements**:

- mostly just grouping common fields, changing descriptions
- adding `JsonFormsUnionRootControl` which handles empty Control block when using `Union` type
- updated links input (`JsonFormsLinkControl`) to use `BaseLinkControl`- which was extracted from `JsonFormsRefControl` - note: this PR does not refactor away this code and there seems to require more work, which isn't the priority at the moment
- refactoring: adding `AltTextSchema` and `ImageSrcSchema` to be reused for consistency of the descriptions and single source of truth

## Tests

please check against the list in the notion of the linear ticket (those labelled 1️⃣ )

### CODE REVIEW

**Comment by adriangohjw** [comment]
@jandonyada fixed storybook

**Review by seaerchin — COMMENTED** [review]
some questions! i think we need to be cognisant in tackling the short term fixes like the link editor shift here - could i trouble you to file a ticket?

File: packages/components/src/interfaces/complex/Infopic.ts [hunk 1]
```diff
@@ -53,6 +41,8 @@ export const InfopicSchema = Type.Object(
         pattern: LINK_HREF_PATTERN,
       }),
     ),
+    imageSrc: ImageSrcSchema,
+    imageAlt: Type.Optional(AltTextSchema),
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply packages/components/src/interfaces/complex/Infopic.ts:45]: it does, and it's intentional based on [SY's comment in Notion](https://www.notion.so/opengov/Studio-Components-editing-janks-14677dbba78880ffb4ceff976d669dbe?pvs=4#15277dbba78880b687ecc41c4028f28f)

File: apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsUnionRootControl.tsx
[the remark this answers is not in the document]
Comment (adriangohjw) [reply apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsUnionRootControl.tsx:1]: the styling is actually not coming from `AllOf/AnyOf` control directly, but that they may render a empty array of non-referencing control object `{ type: "Control", scope: "#" }`

there's nothing to render for that and thus they will render an empty div, which in turn cause the gap to happen.

this is to explicitly catch that and tell it not to render the div

i believe it's slightly similar to how `JsonFormsConstControl` works

**Comment by adriangohjw** [comment]
add a link to refactor: https://linear.app/ogp/issue/ISOM-1703/refactor-link-editor

**Review by seaerchin — APPROVED** [review]
lgtm w hte minor tech debt introduced for links but we should get around to fixing it soon

**Pull request merged**